subprojects/build-cache/src/main/java/org/gradle/caching/local/internal/DirectoryBuildCacheServiceFactory.java

```diff
@@ -81,14 +84,15 @@ public BuildCacheService createBuildCacheService(DirectoryBuildCache configurati
         checkDirectory(target);
 
         int removeUnusedEntriesAfterDays = configuration.getRemoveUnusedEntriesAfterDays();
+        Supplier<Long> removeUnusedEntriesTimestamp = TimestampSuppliers.daysAgo(removeUnusedEntriesAfterDays);
         describer.type(DIRECTORY_BUILD_CACHE_TYPE).
             config("location", target.getAbsolutePath()).
             config("removeUnusedEntriesAfter", String.valueOf(removeUnusedEntriesAfterDays) + " days");
 
         PathKeyFileStore fileStore = fileStoreFactory.createFileStore(target);
         PersistentCache persistentCache = cacheRepository
             .cache(target)
-            .withCleanup(createCacheCleanup(removeUnusedEntriesAfterDays))
+            .withCleanupStrategy(createCacheCleanupStrategy(removeUnusedEntriesTimestamp))
```

Comment:
Yeah, there are several things that have to be specified here, but they're not really boilerplate.  Can you elaborate on what we would want to make better here?  Or rather, what part of constructing these various types do you feel is sub-optimal?

I did play around with both of your suggestions and I don't feel like they were fundamentally better (or necessarily worse).  It just sort of moved things around.  I think we could potentially make `DefaultCacheCleanupStrategy` go away by adding another method to `CacheBuilder`, but again, it didn't clearly make things better.  If there's some aspect of this that we specifically want to improve, it might be easier to focus on that.

Reply:
Nothing specific in mind, I just thought it would maybe look nicer done slightly differently, hence the Monocle.  Thanks for giving it a try.